modules/tinymce/src/core/main/ts/api/html/SaxParser.ts

```diff
@@ -119,38 +119,60 @@ const isInvalidUri = (settings: SaxParserSettings, uri: string, tagName: string)
 };
 
 /**
- * Returns the index of the end tag for a specific start tag. This can be
- * used to skip all children of a parent element from being processed.
+ * Returns the index of the matching end tag for a specific start tag. This can
+ * be used to skip all children of a parent element from being processed.
  *
  * @private
- * @method findEndTagIndex
+ * @method findMatchingEndTagIndex
  * @param {tinymce.html.Schema} schema Schema instance to use to match short ended elements.
  * @param {String} html HTML string to find the end tag in.
  * @param {Number} startIndex Index to start searching at should be after the start tag.
  * @return {Number} Index of the end tag.
  */
-const findEndTagIndex = (schema: Schema, html: string, startIndex: number): number => {
-  let count = 1, index, matches;
-
+const findMatchingEndTagIndex = (schema: Schema, html: string, startIndex: number): number => {
+  // TODO: TINY-7658: this regex does not support CDATA
+  const startTagRegExp = /<([!?\/])?([A-Za-z0-9\-_:.]+)/g;
+  const endTagRegExp = /(?:\s(?:[^'">]+(?:"[^"]*"|'[^']*'))*[^"'>]*(?:"[^">]*|'[^'>]*)?|\s*|\/)>/g;
   const shortEndedElements = schema.getShortEndedElements();
-  const tokenRegExp = /<([!?\/])?([A-Za-z0-9\-_:.]+)(\s(?:[^'">]+(?:"[^"]*"|'[^']*'))*[^"'>]*(?:"[^">]*|'[^'>]*)?|\s*|\/)>/g;
-  tokenRegExp.lastIndex = index = startIndex;
+  let count = 1, index = startIndex;
+
+  // keep finding HTML tags (opening, closing, or neither like comments or <br>s
```

Comment:
```suggestion
  // keep finding HTML tags (opening, closing, or neither like comments or <br>s)
```
There's a missing end parenthesis here. I'm assuming this is where you meant it to be.